metadata-ingestion/src/datahub/ingestion/source/aws/glue.py

```diff
@@ -790,6 +798,49 @@ def get_dataset_properties() -> DatasetPropertiesClass:
                 tags=[],
             )
 
+        def get_s3_tags() -> GlobalTagsClass:
+            bucket_name = s3_util.get_bucket_name(
+                table["StorageDescriptor"]["Location"]
+            )
+            tags_to_add = []
+            if self.source_config.use_s3_bucket_tags:
+                bucket_tags = self.s3_client.get_bucket_tagging(Bucket=bucket_name)
+                tags_to_add.extend(
+                    [
+                        make_tag_urn(f"""{tag["Key"]}:{tag["Value"]}""")
+                        for tag in bucket_tags["TagSet"]
+                    ]
+                )
+            if self.source_config.use_s3_object_tags:
+                key_prefix = s3_util.get_key_prefix(
+                    table["StorageDescriptor"]["Location"]
+                )
+                object_tagging = self.s3_client.get_object_tagging(
+                    Bucket=bucket_name, Key=key_prefix
+                )
+                tags_to_add.extend(
+                    [
+                        make_tag_urn(f"""{tag["Key"]}:{tag["Value"]}""")
+                        for tag in object_tagging["TagSet"]
+                    ]
+                )
+            if self.ctx.graph is not None:
+                current_tags: Optional[GlobalTagsClass] = self.ctx.graph.get_aspect_v2(
+                    entity_urn=dataset_urn,
+                    aspect="globalTags",
+                    aspect_type=GlobalTagsClass,
+                )
+                if current_tags:
+                    tags_to_add.extend(
+                        [current_tag.tag for current_tag in current_tags.tags]
+                    )
```

Comment:
Do we skip updating tags aspect when tags_to_add is empty? 
Also, If the configs to add tags is set in source config but tags could not be added due to  self.ctx.graph being None, we should stop execution with configuration error. similar to this - https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/src/datahub/ingestion/transformer/add_dataset_ownership.py#L64

Reply:
This also applies for s3 source.

Reply:
Changed to not add tags to aspect if they are empty.  Also added that ctx.graph configuration error.

Reply:
@Jiafi `ctx.graph = None` does not represent that there are no tags to maintain, the graph is none if using sink other than datahub-rest( like file or datahub kafka sink), in which case datahub_api needs to be explicitly set in recipe file to be able to instantiate ctx.graph.

With current code, if older tags were present and datahub kafka sink it used, tags will be overridden.

I understand your concern that someone who has not setup datahub and uses file sink for testing (and not for ingesting file to datahub later) might end up with ConfigurationError with suggested changes. Maybe, we can continue with just a warning message like you suggested or take additional config whether to ignore ctx unavailable error.

Reply:
@Jiafi `ctx.graph = None` does not represent that there are no tags to maintain, the graph is none if using sink other than datahub-rest( like file or datahub kafka sink), in which case datahub_api needs to be explicitly set in recipe file to be able to instantiate ctx.graph.

With current code, if older tags were present and datahub kafka sink it used, tags will be overridden.

I understand your concern that someone who has not setup datahub and uses file sink for testing (and not for ingesting file to datahub later) might end up with ConfigurationError with suggested changes. Maybe, we can continue with just a warning message like you suggested or take additional config whether to ignore ctx unavailable error.

Reply:
I personally think a warning is correct call.  If there extra configurations for the source to tell that its using a datahub api, it can very easily get disjointed with actually using the sink.  Theres more room for configuration errors as opposed to the warning message being pretty clear about what is going on.